### DISCUSSION TITLE
threshold and kernel-size, and the limitation of odd-integer kernel-size

### DISCUSSION THREAD

**Discussion by ceewanna** [Breakthrough/DVR-Scan#121] [General]
After running through a number of sample video events, trying to fine tune parameters, I have some observations on the 2 parameters.

Currently, I have 2 sets of parameters.
1. threshold 0.15, kernel-size 11
2. threshold 0.3, kernel-size 9

Other parameters used include
      min-event-length = 0.2s
      time-before-event = 3s
      time-post-event = 7s
      downscale-factor = 2
      frame-skip = 2

Camera/Video resolution : 2560x1440

My initial thought was that the first set would be more sensitive to motion. However, when it was really put into to operations. The second set created was more sensitive to noise.

After many round of testing, I tend to believe that kernel-size would handle noise. However, the kernel-size is limited to odd integer, it's very more difficult to fine tune the overall function. In this situation, switching between 9 and 11 creates substantially different outcomes.

Right now, how I handle noise, e.g. subtle movement of trees or night time twinkle lights, I would jag up kernel-size one more step and then lower the threshold otherwise it wouldn't be able to detect what was detectable at the lower kernel-size. This would counter intuition as threshold/255 indicates sensitivity https://github.com/Breakthrough/DVR-Scan/discussions/74#discussion-3842638. 

Considering my hands-on testing, threshold might not be the indicator for sensitivity and I hope there would be room for adjusting kernel-size.

**Comment by Breakthrough** [comment]
Sorry for the delay in responding to this.  Unfortunately kernel size *must* be an odd integer due to the way this filter works.  It requires an odd-sized matrix to be applied.  Right now it uses a morphological filter to reduce noise, but there are more expensive techniques (e.g. blurring) that can achieve much finer grained noise reduction at the expense of performance.
